src/main/java/org/openstreetmap/atlas/geography/atlas/BareAtlas.java

```diff
@@ -914,6 +942,34 @@ private void checkRelationMembersAndIndexRelation(final Relation relation,
         }
     }
 
+    private <Member extends AtlasEntity> Supplier<Iterable<Member>> getCachingSupplier(
+            final Iterable<Member> source, final ItemType type)
+    {
+        final List<Set<Long>> memberIdentifiersIntersecting = new ArrayList<>();
```

Comment:
Could this also work as an `Optional`? Might make things a little clearer, since it's a singleton `List` anyway.

Reply:
Well it would not work as an optional given the fact that you cannot swap the internals of an Optional object without changing the optional handle... or can you?
Basically, like in https://github.com/osmlab/atlas/pull/251/files#r227476980, the issue is that I can share objects within the lambda only when they are final. Open to other ideas though!

Reply:
Yeah you can't, so that wouldn't work.... Darn